com.unity.multiplayer.mlapi/Runtime/Core/NetworkManager.cs

```diff
@@ -1486,5 +1476,34 @@ internal void HandleApproval(ulong clientId, bool createPlayerObject, ulong? pla
                 NetworkConfig.NetworkTransport.DisconnectRemoteClient(clientId);
             }
         }
+
+        private void ProfilerBeginTick()
+        {
+            PerformanceDataManager.BeginNewTick();
+            if (NetworkConfig.NetworkTransport is ITransportProfilerData profileTransport)
+            {
+                profileTransport.BeginNewTick();
+            }
+        }
+
+        private void NotifyProfilerListeners()
+        {
+            var data = PerformanceDataManager.GetData();
+            var eventHandler = OnPerformanceDataEvent;
+            if (eventHandler != null && data != null)
+            {
+                if (NetworkConfig.NetworkTransport is ITransportProfilerData profileTransport)
+                {
+                    var transportProfilerData = profileTransport.GetTransportProfilerData();
+                    PerformanceDataManager.AddTransportData(transportProfilerData);
+                }
+
+                eventHandler.Invoke(data);
+            }
+            else if (data == null)
+            {
+                NetworkLog.LogWarning("No data available. Did you forget to call PerformanceDataManager.BeginNewTick() first?");
```

Comment:
Some suggestions for this log, in case it makes its way to users:
- "No data available" should be more specific. "No performance data available." perhaps
- "Did you forget to call..." this will confuse users because it's not something they are supposed to call. At least i think it should be made sure that this is an internal error, and not something they can necessarily fix

Reply:
minor: I'd also suggest string interpolation syntax and `nameof` operator here.

`NetworkLog.LogWarning($"No data available. Did you forget to call {nameof(PerformanceDataManager)}.{nameof(BeginNewTick)}() first?");`

Reply:
Agreed, that's worth a quick fix to use the future-proofed naming system.  We've started to adopt that make make it our standard.